pkg/kubectl/cmd/cp/cp_test.go

```diff
@@ -580,6 +580,42 @@ func TestClean(t *testing.T) {
 	}
 }
 
+func TestConvertHostDestFilename(t *testing.T) {
+	tests := []struct {
+		srcFile           string
+		destFile          string
+		convertedFilename string
+	}{
+		{
+			"test_convert_host_file",
+			"test_convert_host_dir",
+			"test_convert_host_dir/test_convert_host_file",
+		},
+	}
+	for _, test := range tests {
+		err := os.Mkdir(test.destFile, 755)
```

Comment:
It's better to create a temporary directory `ioutil.TempDir` at the beginning of the test and then just clean it. See the first lines of `TestTarUntarWrongPrefix` https://github.com/kubernetes/kubernetes/blob/8dea3310e532fcb28eed84c2bb9a16a6cb7cbfdf/staging/src/k8s.io/kubectl/pkg/cmd/cp/cp_test.go#L397-L406